js/__internal/grids/grid_core/columns_controller/module_utils.ts

```diff
@@ -561,6 +561,11 @@ export const updateColumnChanges = function (that, changeType, optionName?, colu
     optionNames.length++;
   }
   if (columnIndex === undefined || columnIndex !== columnChanges.columnIndex) {
+    if (isDefined(columnIndex)) {
+      columnChanges.columnIndices = columnChanges.columnIndices || [columnChanges.columnIndex];
+      columnChanges.columnIndices.push(columnIndex);
+    }
+
```

Comment:
Let's make this part of the code more obvious:

```suggestion
    if(!isDefined(columnChanges.columnIndices)) {
      columnChanges.columnIndices = [columnChanges.columnIndex];
    }
    
    if (isDefined(columnIndex)) {
      columnChanges.columnIndices.push(columnIndex);
    }
```